pkg/io/mdWriter.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package io
+
+import (
+	"bufio"
+	"io"
+	"slices"
+	"strings"
+
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/ir"
+)
+
+const (
+	sgColsNum  = 7
+	aclColsNum = 10
+
+	leftAlign = " :--- "
+	separator = " | "
+)
+
+// MDWriter implements ir.Writer
+type MDWriter struct {
+	w *bufio.Writer
+}
+
+func NewMDWriter(w io.Writer) *MDWriter {
+	return &MDWriter{w: bufio.NewWriter(w)}
+}
+
+func (w *MDWriter) WriteSG(collection *ir.SGCollection, vpc string) error {
+	sgTable, err := WriteSG(collection, vpc)
+	if err != nil {
+		return err
+	}
+	return w.writeAll(slices.Concat(SGHeader(), addAligns(sgColsNum), sgTable))
```

Comment:
```suggestion
    sgHeader := SGHeader()
	return w.writeAll(slices.Concat(sgHeader, addAligns(len(sgHeader[0])), sgTable))
```
Seems safer if we decide to change the number of columns